src/data/option/basic.lean

```diff
@@ -142,4 +142,25 @@ theorem lift_or_get_choice {f : α → α → α} (h : ∀ a b, f a b = a ∨ f
 | none     (some b) := or.inr rfl
 | (some a) (some b) := by simpa [lift_or_get] using h a b
 
+/-! Add functorial coercions and basic lemmas about it. -/
+section coercions
+variable [has_coe α β]
+
+instance option.to_option : has_coe (option α) (option β) := ⟨option.map (λ a, (a : β))⟩
+
+@[simp, move_cast]
+lemma option.coe_some (a : α) : (↑(some a : option α) : option β) = some ↑a := rfl
+
+@[simp, squash_cast]
+lemma option.coe_none : (↑(none : option α) : option β) = (none : option β) := rfl
+
+@[simp]
```

Comment:
```suggestion
```
This shouldn't be a simp-lemma. It destroys the canonical representation. Maybe if you phrase it the other way round...

Reply:
What is the "canonical representation" in this case? My canonical representation, for example, of the value `(7 : option nat)` would be `some 7` rather than the former.

Reply:
Then why would introduce all the coercions in the first place?

Reply:
Hmm? I did not introduce the coercion from `α` to `option α`. I am just adding simplification lemmas so that things become more automatic.